src/components/Contacts/ContactDetails/ContactDetailsHeader/DeleteContactModal/DeleteContactModal.tsx

```diff
@@ -12,37 +17,178 @@ import {
   DeleteButton,
 } from 'src/components/common/Modal/ActionButtons/ActionButtons';
 import Modal from 'src/components/common/Modal/Modal';
+import { useAccountListId } from 'src/hooks/useAccountListId';
+import { isEditableSource, sourceToStr } from 'src/utils/sourceHelper';
+import { useContactSourceQuery } from './ContactSource.generated';
 
 const LoadingIndicator = styled(CircularProgress)(({ theme }) => ({
   margin: theme.spacing(0, 1, 0, 0),
 }));
 
+interface DataInfo {
+  canDeleteWithoutIssues: boolean;
+  contactSource: string;
+  addressSources: string[];
+  emailSources: string[];
+  phoneSources: string[];
+}
 interface DeleteContactModalProps {
   open: boolean;
   setOpen: (open: boolean) => void;
   deleting: boolean;
   deleteContact: () => void;
+  contactId: string;
 }
 
 export const DeleteContactModal: React.FC<DeleteContactModalProps> = ({
   open,
   setOpen,
   deleting,
   deleteContact,
+  contactId,
 }) => {
   const { t } = useTranslation();
+  const accountListId = useAccountListId() ?? '';
+
+  const { data } = useContactSourceQuery({
+    variables: { accountListId, contactId },
+    skip: !open && !contactId,
+  });
+  const contactSources = data?.contact;
+
+  const dataInfo: DataInfo = useMemo(() => {
+    if (!contactSources) {
```

Comment:
I don't think we need a special case when `contactSources` is undefined. Just change the loops to `contactSources?.field` and no sources will be added when `contactSources` is undefined.